apps/src/lab2/views/components/Instructions.tsx

```diff
@@ -134,22 +147,37 @@ const InstructionsPanel: React.FunctionComponent<InstructionsPanelProps> = ({
   message,
   messageIndex,
   imageUrl,
-  showNextButton,
+  showContinueButton,
+  showFinishButton,
+  beforeNextLevel,
   onNextPanel,
   layout = 'vertical',
   imagePopOutDirection = 'right',
   theme = 'dark',
   handleInstructionsTextClick,
 }) => {
   const [showBigImage, setShowBigImage] = useState(false);
+  const [isFinished, setIsFinished] = useState(false);
 
   const imageClicked = () => {
     setShowBigImage(!showBigImage);
   };
 
   const vertical = layout === 'vertical';
 
-  const canShowNextButton = showNextButton && onNextPanel;
+  const canShowContinueButton = showContinueButton && onNextPanel;
+
+  const canShowFinishButton = showFinishButton;
+
+  const onFinish = () => {
+    if (beforeNextLevel) {
+      beforeNextLevel();
+    }
+    setIsFinished(true);
+  };
+
+  const finalMessage =
+    '<strong>You finished this lesson! Check in with your teacher for the next activity.</strong>';
```

Comment:
Does this need to be translated?

Reply:
This is a temporary UI solution and will most likely be iterated on in the next couple weeks. We wanted something in place for the pilot so I think maybe for now, keeping as a constant makes sense until we land on a more permanent solution. Thanks for bring up!